var/spack/repos/builtin/packages/scalasca/package.py

```diff
@@ -38,29 +38,35 @@ class Scalasca(Package):
     homepage = "http://www.scalasca.org"
     url = "http://apps.fz-juelich.de/scalasca/releases/scalasca/2.1/dist/scalasca-2.1.tar.gz"
 
-    version('2.2.2', '2bafce988b0522d18072f7771e491ab9',
-            url='http://apps.fz-juelich.de/scalasca/releases/scalasca/2.2/dist/scalasca-2.2.2.tar.gz')
-
-    version('2.1', 'bab9c2b021e51e2ba187feec442b96e6',
-            url='http://apps.fz-juelich.de/scalasca/releases/scalasca/2.1/dist/scalasca-2.1.tar.gz')
+    version('2.3.1', 'a83ced912b9d2330004cb6b9cefa7585')
+    version('2.2.2', '2bafce988b0522d18072f7771e491ab9')
+    version('2.1',   'bab9c2b021e51e2ba187feec442b96e6')
 
     depends_on("mpi")
     ##########
     # Hard-code dependencies for Scalasca according to what stated in the
     # release page
     # The OTF2 library path should be detected automatically from SCOREP
     # SCALASCA 2.2.2
-    depends_on("scorep@1.4:", when='@2.2.2')
-    depends_on("cube@4.3:", when='@2.2.2')
+    depends_on("scorep@1.4:", when='@2.2.2:')
+    # Use same cube as scorep does
+    # -> depends_on("cube@4.3:", when='@2.2.2:')
     # SCALASCA 2.1
     depends_on("scorep@1.3", when='@2.1')
-    depends_on("cube@4.2:", when='@2.1')
+    # Use same cube as scorep does
+    # -> depends_on("cube@4.2:", when='@2.1')
     ##########
 
-    def install(self, spec, prefix):
-        configure_args = ["--prefix=%s" % prefix,
-                          "--with-cube=%s" % spec['cube'].prefix.bin,
-                          "--enable-shared"]
-        configure(*configure_args)
-        make()
-        make("install")
+    def url_for_version(self, version):
+        return 'http://apps.fz-juelich.de/scalasca/releases/scalasca/' + \
+            '{0}/dist/scalasca-{1}.tar.gz'.format(version.up_to(2), version)
```

Comment:
You don't actually need to split this line into two. We automatically exempt lines containing `https?:` from the flake8 line length checks.